dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java

```diff
@@ -0,0 +1,756 @@
+        } catch (Exception e) {
+            throw new TestException("Following error occurred while executing " + e.getMessage());
+        }
+        return scriptPath;
+    }
+
+    protected String getUserAppJarLocation(String userAppJar) {
+        String jarPath = null;
+        try {
+            jarPath = hd.getGemFireHome() + sep + ".." + sep + ".." + sep + ".." + sep + "dtests" + sep + "build-artifacts" + sep + "scala-2.10" + sep + "libs" + sep + userAppJar;
+        } catch (Exception e) {
+            throw new TestException("Following error occurred while executing " + e.getMessage());
+        }
+        return jarPath;
+    }
+
+    public static void HydraTask_initializeSnappyTest() {
+        if (snappyTest == null) {
+            snappyTest = new SnappyTest();
+            snappyTest.generateConfig("locators");
+            snappyTest.generateConfig("servers");
+            snappyTest.generateConfig("leads");
+        }
+    }
+
+    public static synchronized void HydraTask_generateSnappyConfig() {
+        locatorsFilePath = snappyTest.getSnappyPath() + sep + "conf" + sep + "locators";
+        serversFilePath = snappyTest.getSnappyPath() + sep + "conf" + sep + "servers";
+        leadsFilePath = snappyTest.getSnappyPath() + sep + "conf" + sep + "leads";
+        try {
+            String dirPath = snappyTest.getLogDir();
+            File locatorsFile = new File(locatorsFilePath);
+            File serversFile = new File(serversFilePath);
+            File leadsFile = new File(leadsFilePath);
+            if (dirPath.contains("locator")) {
+                String locatorLogDir = localHost + " -dir=" + dirPath;
+                if (!locatorsFileContent.contains(locatorLogDir)) {
+                    locatorsFileContent.add(locatorLogDir);
+                    snappyTest.writeToFile(locatorLogDir, locatorsFile);
+                }
+            } else if (dirPath.contains("Store") || dirPath.contains("server")) {
+                String serverLogDir = localHost + " -dir=" + dirPath;
+                if (!serversFileContent.contains(serverLogDir)) {
+                    serversFileContent.add(serverLogDir);
+                    snappyTest.writeToFile(serverLogDir, serversFile);
+                }
+            } else if (dirPath.contains("lead")) {
+                String leadLogDir = localHost + " -dir=" + dirPath;
+                if (leadHost == null) {
+                    leadHost = localHost;
+                }
+                Log.getLogWriter().info("Lead host is: " + leadHost);
+                if (!leadsFileContent.contains(leadLogDir)) {
+                    leadsFileContent.add(leadLogDir);
+                    snappyTest.writeToFile(leadLogDir, leadsFile);
+                }
+            }
+        } catch (Exception e) {
+            throw new TestException("Following error occurred while executing " + e.getStackTrace());
+        }
+    }
```

Comment:
the above method can be re factored as 

  public static synchronized void HydraTask_generateSnappyConfig() {
        sqlTest.generateSnappyConfig()
    }

    
    public generateSnappyConfig() {
       String  confDir =  snappyTest.getSnappyPath() + sep + "conf" + "sep";
       String dirPath = snappyTest.getLogDir();
       String LogFileContent =  HostHelper.getLocalHost() + " -dir=" + dirPath;
       String logFile =""
           if (dirPath.contains("locator"))
                logFile = confDir + "locators";
           elseif (dirPath.contains("server"))
                logFile = confDir + "servers";
           elseif (dirPath.contains("leads)"))
                logFile = confDir + "leads";
           else
             throw new Exception("Invalid Directory path found " + dirPath )
        writeToFile(leadLogDir, leadsFile)

    }


    protected void writeToFile(String logDir, String filePath) {
        try {
            FileWriter fw = new FileWriter(filePath, true);
            BufferedWriter bw = new BufferedWriter(fw);
            bw.write(logDir);
            bw.newLine();
            bw.close();
        } catch (IOException e) {
            throw new TestException("Following error occurred while executing " + e.getMessage());
        }
    }